### ISSUE TITLE
Message Payload required or not

### ISSUE THREAD

**Issue by stereomon** [asyncapi/spec#928]
I already asked this question in the Slack Channel for the Specification and was asked to clarify it with an issue here.

https://www.asyncapi.com/docs/reference/specification/v2.2.0#definitionsMessage (The statement is the same for all higher versions)

It states that a message `MUST` have a payload. But what if I have a command message with the headers containing all the data I need? Do I really need to duplicate the data in the payload?

@derberg mentioned that it is confusing that on the message definition level, the spec says `MUST` and in the https://www.asyncapi.com/docs/reference/specification/v2.2.0#messageObject it is not marked as `REQUIRED`.

From my perspective, it shouldn't be a `MUST` and it shouldn't be `REQUIRED` to have a payload defined in each message specifically not when the headers contain all relevant data for me.

Is the documentation wrong or do I miss something?

**Comment by derberg** [comment]
For me it is just an editorial error.
Whatever is in definition
<img width="1040" alt="Screenshot 2023-04-26 at 14 36 44" src="https://user-images.githubusercontent.com/6995927/234576852-b730ad07-5f4c-4ea7-ad8f-a94caa0227ba.png">

Is overwritten by Message Object definition that do not `REQUIRE` payload
![Screenshot 2023-04-26 at 14 38 59](https://user-images.githubusercontent.com/6995927/234578731-e80a2019-f8b7-4800-9320-d6b79d2383a8.png)


Also schema do not require it either: https://github.com/asyncapi/spec-json-schemas/blob/master/schemas/2.6.0.json#L894

---
My proposal is to say `MAY` in case of message payload in the definition section


@char0n @fmvilas @smoya @dalelane thoughts?

**Comment by dalelane** [comment]
Would the absence of a message payload definition imply "these messages have no payload"? Or simply that the payload is undefined?

If this ambiguity is unhelpful, then there might be value in keeping payload as required, but having a way of explicitly specifying it can be empty.

**Comment by stereomon** [comment]
> Would the absence of a message payload definition imply "these messages have no payload"?
Yes, messages having all needed data specified in the header for processing doesn't need a payload.

Making it a required field doesn't make sense from my point of view. Not sure how e.g. Kafka would react to a non present payload.

I would prefer to have this as an optional field in the specification.

**Comment by dalelane** [comment]
Sorry, I didn't explain myself clearly enough. 

I wasn't questioning whether different messaging/streaming technologies tolerate the absence of a payload. 

I was considering and discussing the best way to document and describe a message without a payload. Specifically, I was comparing one approach (that omitting the payload section from the doc specifies that there is no payload) with another approach (having a payload section in the AsyncAPI doc, and in that section you can explicitly say that the payload is empty)

**Comment by derberg** [comment]
we are having the same situation as with headers, if headers are not provided, does it mean they are unknown or that there are no headers?

**Comment by dalelane** [comment]
That's true... my assumption would be that if something isn't specified, then it's unknown - that the most likely reason is that the author didn't think to specify it. And that's certainly my assumption with something like headers which are often not documented. 

I'm not sure I'm strongly arguing for this, more thinking out loud... I was considering something along the lines of authors needing to explicitly specify something like `null` to specify that messages actually have no payload. But I think you're right that if we did that, then the same could/should apply to headers, too.

**Comment by char0n** [comment]
I would suggest to make an editorial change to replace the `MUST` with `MAY` word to remove ambiguity from the specification and to make things right.

The source of truth for me (tooling author) are actual objects definitions (Message Object, Channel Object) which supersedes everything else within the specification text. Maybe it would be worth mentioning this inside the spec itself? We're already dealt with the source of truth before in https://github.com/asyncapi/spec/pull/781.

Question about the actual clarity is probably for another issue. For me personally omitting the `payload` means that my Message Object `MAY` have payload but I didn't care describing it. Given it's not possible to describe an absence of a value by the JSON Schema, the clarify is implicit but clear (correct me if I'm wrong).

**Comment by char0n** [comment]
@smoya editorial change issued as https://github.com/asyncapi/spec/pull/983
